### PR TITLE
Update links for GeoMoose 3 docs.

### PR DESCRIPTION
**Pull request by klassenjs**
* Remove outdated GeoMoose 2 references.
* Update GM3 links from GitHub Pages site to new Sphinx site.
* Use `main` branch for GeoMoose 3.

### CODE REVIEW

**Comment by klassenjs** [comment]
Updated version of #14

**Comment by klassenjs** [comment]
Note: this includes updated links and minor wording for RFC-4.  I'm not sure what the protocol is for updating RFCs.

File: source/rfc/rfc-4.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ RFC 4: GeoMoose Commit Management
 :Contact: bob DOT basques at ci DOT stpaul DOT mn DOT us
 :Contact: klassen DOT js at gmail DOT com
 :Status: Adopted
-:Last Updated: 2013-07-25
+:Last Updated: 2021-02-05
```
Comment (elil) [inline source/rfc/rfc-4.rst:11]: I would probably leave the date as 2013-07-25 and link to this housekeeping commit.  Or possibly list both dates.  Oh, I see you more or less already address this at the bottom.

**Comment by elil** [comment]
I didn't build the docs but these all look like reasonable updates.

File: source/rfc/rfc-4.rst (hunk 1 above)
Comment (theduckylittle) [reply source/rfc/rfc-4.rst:11]: Agree with Eli, as we are only scrubbing toxic language, I believe we can keep the date the same.

File: source/rfc/rfc-4.rst [hunk 2]
```diff
@@ -169,3 +169,7 @@ Voting History
 --------------
 Adopted on 6/27/2013 with PSC +1 votes from Eli Adam, Bob Basques, Brian Fischer, 
 Brent Fraser, Jim Klassen.
+
+Updates
+-------
+2021-02-05: Update links for GeoMoose 3
```
Comment (theduckylittle) [inline source/rfc/rfc-4.rst:175]: This makes sense. The update log to just note the changes.

**Review by theduckylittle — APPROVED** [review]
Approved with minor note.

**Pull request merged**